### ISSUE TITLE
Feature request : UPS monitoring with apcupsd

### ISSUE THREAD

**Issue by lbalogh** [henrygd/beszel#355]
Hi,

First of all, do not hesitate to close this demand if it's out of scope, I would totally understand.

A few days ago, I was playing with Netdata, and I noticed that it magically shows statitistics for my UPS, which is managed by the apcupsd daemon on my host. There are graphs for current consumption (in W), minutes left, battery percentage, etc.

It would be very cool to have this kind of stats in Beszel ;)

Here is the Netadata documentation for reference : https://learn.netdata.cloud/docs/collecting-metrics/ups/apc-ups

**Comment by henrygd** [comment]
I agree this would be a nice feature. I'll look into it. Would be nice if there was an open protocol though, rather than an APC specific thing.

We're also planning to add custom charts where you can supply data via command or file content, so it should be possible in the future even if direct support isn't added.

**Comment by playaz44** [comment]
> I agree this would be a nice feature. I'll look into it. Would be nice if there was an open protocol though, rather than an APC specific thing.
> 
> We're also planning to add custom charts where you can supply data via command or file content, so it should be possible in the future even if direct support isn't added.

Hello, you can check NUT [https://networkupstools.org/](https://networkupstools.org/), it has home assistant integration [https://www.home-assistant.io/integrations/nut/](https://www.home-assistant.io/integrations/nut/).

**Comment by lbalogh** [comment]
> I agree this would be a nice feature. I'll look into it. Would be nice if there was an open protocol though, rather than an APC specific thing.
> 
> We're also planning to add custom charts where you can supply data via command or file content, so it should be possible in the future even if direct support isn't added.

Thanks. Custom charts will be amazing !

**Comment by snapshotleisure** [comment]
Hi @henrygd, This would be a nice a feature, one suggestion would be to piggy back off another open source project like peaNUT. https://github.com/Brandawg93/PeaNUT

They have already spent the effort on nuts and make it work with multiple UPS's and different brands. They also have an API which you can just call to populate the data. This might be a quick win.

The downside will be that people need to add another container, but I think this is fine as only people with UPS will need to install PeaNUT.

I only suggest this to try and reduce the work to support  this feature and also maintenance. I worry with all these feature request that you might get overwhelmed, especially with trying to implement them natively. I think trying to integrate with other FOSS tools out there might be a good way to go.

**Comment by healy93** [comment]
I agree NUT would be great!

**Comment by henrygd** [comment]
Makes sense. Added it to the roadmap.

**Comment by mrvalterhugo** [comment]
+1 This would be awesome, I currently use Nutify, which allows to even see the calculated cost.

**Comment by leon-ITA** [comment]
+1 would be great have the UPS information with nut

**Comment by mrvalterhugo** [comment]
I have been using https://github.com/DartSteven/Nutify in my homelab, would be nice having something like this in beszel.

**Comment by manu2107el** [comment]
There is an ongoing discussion on the UPS Topic: https://github.com/henrygd/beszel/discussions/1613
